### PR TITLE
Ctools 4

### LINKED ISSUE #7395: Update ctools and ctools views
# Proposed Solution

- Update Ctools module to most recent version.
- Test to make sure there are no regressions with existing views blocks displays.
- Address any regressions found.

### PR DESCRIPTION
**Pull request by pyrello**
Resolves #7395 

I thought I was being clever before when I made sure that our views block display allow setting had the same key name (`configure_filters`) as the feature in Ctools Views. However, this turned out to be a mistake because it meant it was impossible to leave our filters turned on and keep the new filters turned off. To get around this, I changed the key name for our filters to `configure_filters_custom`.  This allowed us to keep the new feature in the module turned off and our existing custom feature turned on.

As such, I ended up going through and re-saving all the views that contained this key to make sure I was picking up all the relevant changes. Apparently, if you don't click into the allow settings and you save a view, it doesn't include them in the display's settings in config. I specifically clicked the allow settings in every views block display to make sure they were up-to-date and consistent. This means that some of the views config seemingly contains new `allow` settings for some block displays. Examining these should reveal that everything is "turned off".

# How to test

- Review [file changes](https://github.com/uiowa/uiowa/pull/7396/files).
- Test article and people list blocks to ensure no functionality has changed.
- Test as many of the custom list blocks as is necessary to convince yourself that nothing has changed.

### CODE REVIEW

File: config/sites/admissions.uiowa.edu/views.view.majors.yml [hunk 1]
```diff
@@ -1079,6 +1079,23 @@ display:
...
+        disable_filters: '0'
+        configure_sorts: '0'
+        configure_filters_custom: 0
+        use_more: 0
+      general_help_text: ''
+      more_link_help_text: ''
+      restrict_fields:
+        nid: 0
+        field_major_area_of_study: 0
+        title: 0
```
Comment (pyrello) [inline config/sites/admissions.uiowa.edu/views.view.majors.yml]: Hmm... not sure why this was changed. Need to look into this.

Comment (pyrello) [reply config/sites/admissions.uiowa.edu/views.view.majors.yml]: See the explanation above the test instructions above.

**Comment by joewhitsitt** [comment]
Going to pull down and start testing

**Comment by joewhitsitt** [comment]
Something isn't quite right. Comparing fresh sync of https://admissions.uiowa.ddev.site/ and https://admissions.uiowa.edu/

Screenshot 2024-02-28 at 10 15 33 AM

**Review by joewhitsitt — APPROVED** [review]
I saw https://git.drupalcode.org/project/ctools/-/compare/8.x-3.11...4.0.4?from_project_id=59514&straight=false#afe6a515b5ba38f229b3160becfd68f66bf573fc_106_122

Could this be cleaned up now or do we want to continue the min of 1?
https://github.com/uiowa/uiowa/blob/01aad5c3e7faf095decbb881935ebab132bde594/docroot/modules/custom/layout_builder_custom/src/Plugin/Display/ListBlock.php#L163

Do you want to turn https://github.com/uiowa/uiowa/issues/7395#issuecomment-1964819547 into a separate issue?

Should this be a scheduled deployment to avoid issues before CIM is run for the site? See the module's filters are showing at the top of the block.

Screenshot 2024-02-28 at 11 20 21 AM

**Pull request merged**